loki/transformations/remove_code.py

```diff
@@ -355,29 +409,58 @@ class RemoveRegionTransformer(Transformer):
     example via :meth:`pragma_regions_attached`.
 
     When removing a marked code region the transformer may leave a
-    comment in the source to mark the previous location, or remove the
-    code region entirely.
+    comment or a replacement call to trigger "abort" errors in the
+    source to mark the previous location.
 
     Parameters
     ----------
     mark_with_comment : boolean
         Flag to trigger the insertion of a marker comment when
         removing a region; default: ``True``.
+    replacement_call : optional, str
+        Name of the "abort" subroutine to call if a replacement call
+        is to be inserted.
+    replacement_msg : optional, str
+        Optional error message that will be passed as a single
+        argument to the replacmeent call.
     """
 
-    def __init__(self, mark_with_comment=True, **kwargs):
+    def __init__(
+            self, mark_with_comment=True, replacement_call=None, replacement_msg=None, **kwargs
+    ):
         super().__init__(**kwargs)
+
         self.mark_with_comment = mark_with_comment
 
+        # Replace section with call to trigger abort messages!
+        self.replacement_call = replacement_call
+        self.replacement_msg = replacement_msg
+        self.replacement_done = False
+
     def visit_PragmaRegion(self, o, **kwargs):
         """ Remove :any:`PragmaRegion` nodes with ``!$loki remove`` pragmas """
 
         if is_loki_pragma(o.pragma, starts_with='remove'):
             # Leave a comment to mark the removed region in source
+            replacement = []
             if self.mark_with_comment:
-                return Comment(text='! [Loki] Removed content of pragma-marked region!')
-
-            return None
+                replacement.append(ir.Comment(text='! [Loki] Removed content of pragma-marked region!'))
+
+            if self.replacement_call:
+                # Get the outer scope, to avoid picking associates
+                routine = kwargs['scope']
+                while not isinstance(routine, ProgramUnit):
+                    routine = routine.parent
```

Comment:
NB: this line is untested, might be worth adding an associate to one of the tests